### PR TITLE
Block public access by default

### PR DESCRIPTION
**Pull request by chrischildresssg**
## Change description

Add public access blocks for the terraform state bucket, with individual parameters for fine-grained control over blocking public access.  All public access is blocked by default.

### Development

- [x] Lint rules pass locally
- [x] Application changes have been tested thoroughly

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [x] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,12 @@ This is intended to be used by an organization for all of their own accounts. Th
 
 This restriction is put in place by creating a unique role for each account, then attaching an assumerole policy that trusts the corresponding account to assume it. You will still need to give permission to assume roles to your users/roles that are used to apply terraform in other accounts, and configure your state appropriately to use this.
 
+As of v3.0, all public access is blocked by default.  There are individual parameters which can be set to "false" if public bucket/state access is desired:
```
Comment (wesleykirklandsg) [inline README.md:5]: @chrischildresssg Thoughts on a single bool here? Is there a time where we need to toggle each specific one. Another option is a single var toggle with a conditional to determine the logic in it.

File: s3-bucket.tf [hunk 2]
```diff
@@ -124,3 +124,14 @@ resource "aws_s3_bucket_policy" "remote_state_backend" {
   bucket = aws_s3_bucket.remote_state_backend.id
   policy = data.aws_iam_policy_document.encrypted_transit_bucket_policy.json
 }
+
+resource "aws_s3_bucket_public_access_block" "this" {
```
Comment (wesleykirklandsg) [inline s3-bucket.tf:128]: Is this 4.x compatible?

File: README.md [hunk 3]
```diff
@@ -2,6 +2,12 @@ This is intended to be used by an organization for all of their own accounts. Th
...
+* block_public_acls
+* block_public_policy
+* ignore_public_acls
+* restrict_public_buckets
+
 ## Example Config:
```
Comment (wesleykirklandsg) [inline README.md:11]: Can we do TF docs on this README? It would help with the inputs.

**Review by wesleykirklandsg — CHANGES_REQUESTED** [review]
See comments.

File: README.md (hunk 1 above)
Comment (chrischildresssg) [reply README.md:5]: Well, everything is "true" by default. The idea here is that if you specifically needed to allow public ACLs but not the rest, you'd only need to toggle one parameter.  Do you think it makes more sense to have it be "all or nothing"?

File: README.md (hunk 3 above)
Comment (chrischildresssg) [reply README.md:11]: Certainly; I'll get that worked in.

File: README.md (hunk 1 above)
Comment (juansanchez-stratusgrid) [reply README.md:5]: I like having the option to toggle some and not all at the same time, like how it works in the aws console

**Comment by juansanchez-stratusgrid** [comment]
LGTM besides the terraform docs comment from Wesley which I support, I don't know if compatibility with 4.x is crucial atm

File: s3-bucket.tf (hunk 2 above)
Comment (chrischildresssg) [reply s3-bucket.tf:128]: No - the whole module is not, I believe.

**Comment by chrischildresssg** [comment]
@juansanchez-stratusgrid @wesleykirklandsg  please take a look at the updated documentation.  Thanks!

File: README.md (hunk 3 above)
Comment (chrischildresssg) [reply README.md:14]: That's now done; please check the README and let me know what you think.

File: .config/.terraform-docs.yml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  ## Example Backend Config:
+
+  ```hcl
+  {{ include "examples/backend_config.tf" }}
+  ```
+
+  ## Example to Initialize the backend:
+  ```
+  terraform init -backend-config="access_key=AKIAIOIEXAMPLEOXJA" -backend-config="secret_key=PiNMLcNOTAREALKEYGQGzz20v3"
+  ```
```
Comment (juansanchez-stratusgrid) [inline .config/.terraform-docs.yml:35]: I assume this are outdated but i need to ask, is it ok to post access key and secret key on the documentation?

**Review by juansanchez-stratusgrid — COMMENTED** [review]
LGTM, I just have the question with the credentials in the documentation, please let me know if that is ok so i can approve

File: .config/.terraform-docs.yml (hunk 4 above)
Comment (chrischildresssg) [reply .config/.terraform-docs.yml:35]: These have been in place for...a very long time.  Good catch; I'll make them something generic, and I'll ask Ivan if we need to pursue this further.

Comment (wesleykirklandsg) [reply .config/.terraform-docs.yml:35]: Do we need that example? Normally we just leave it as `terraform init -backend-config=` and even then this is a module so it doesn't need this information?

File: examples/no_trusted_accounts.tf [hunk 5]
```diff
@@ -0,0 +1,22 @@
+  module "terraform_state" {
+    source  = "StratusGrid/terraform-state-s3-bucket-centralized-with-roles/aws"
+    version = "2.0.0"
+    # source  = "github.com/StratusGrid/terraform-aws-terraform-state-s3-bucket-centralized-with-roles"
```
Comment (wesleykirklandsg) [inline examples/no_trusted_accounts.tf:4]: This needs to be removed since it's in the TF registry

File: examples/no_trusted_accounts.tf [hunk 6]
```diff
@@ -0,0 +1,22 @@
+  module "terraform_state" {
+    source  = "StratusGrid/terraform-state-s3-bucket-centralized-with-roles/aws"
+    version = "2.0.0"
```
Comment (wesleykirklandsg) [inline examples/no_trusted_accounts.tf:3]: This should be 3.0.0, and all filed should be updated as soo.

File: s3-bucket.tf (hunk 2 above)
Comment (wesleykirklandsg) [reply s3-bucket.tf:128]: Would it be 4.x compatible if it went into the main resource block?

File: examples/no_trusted_accounts.tf (hunk 6 above)
Comment (chrischildresssg) [reply examples/no_trusted_accounts.tf]: @wesleykirklandsg Please see the updated examples and let me know if you see any mistakes.

File: .config/.terraform-docs.yml (hunk 4 above)
Comment (juansanchez-stratusgrid) [reply .config/.terraform-docs.yml:35]: I like it tough, like documentation that is thorough

**Review by juansanchez-stratusgrid — DISMISSED** [review]

File: examples/no_trusted_accounts.tf (hunk 5 above)
Comment (chrischildresssg) [reply examples/no_trusted_accounts.tf]: Check now

File: .config/.terraform-docs.yml (hunk 4 above)
Comment (chrischildresssg) [reply .config/.terraform-docs.yml:35]: For now I'll mark this resolved. If we want to remove the example, we can create an issue for it.

**Comment by wesleykirklandsg** [comment]
Besides some linting issues, I'm happy.

If you could run a TF FMT that would be great, otherwise LGTM.

**Review by wesleykirklandsg — APPROVED** [review]

**Review by wesleykirklandsg — APPROVED** [review]

**Pull request merged**